vm/message/src/message_receipt.rs

```diff
@@ -11,3 +11,72 @@ pub struct MessageReceipt {
     pub return_data: Serialized,
     pub gas_used: u64,
 }
+
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use num_traits::cast::FromPrimitive;
+    use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+    /// Wrapper for serializing and deserializing a SignedMessage from JSON.
+    #[derive(Deserialize, Serialize)]
+    #[serde(transparent)]
+    pub struct MessageReceiptJson(#[serde(with = "self")] pub MessageReceipt);
+
+    /// Wrapper for serializing a SignedMessage reference to JSON.
+    #[derive(Serialize)]
+    #[serde(transparent)]
+    pub struct MessageReceiptJsonRef<'a>(#[serde(with = "self")] pub &'a MessageReceipt);
+
+    impl From<MessageReceiptJson> for MessageReceipt {
+        fn from(wrapper: MessageReceiptJson) -> Self {
+            wrapper.0
+        }
+    }
+
+    pub fn serialize<S>(m: &MessageReceipt, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MessageReceiptSer<'a> {
+            exit_code: u64,
+            #[serde(rename = "Return")]
+            return_data: &'a [u8],
+            gas_used: u64,
+        }
+        MessageReceiptSer {
+            exit_code: m.exit_code as u64,
+            return_data: m.return_data.bytes(),
+            gas_used: m.gas_used,
+        }
+        .serialize(serializer)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<MessageReceipt, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Serialize, Deserialize)]
```

Comment:
```suggestion
        #[derive(Deserialize)]
```
No need at all, doesn't actually matter though